src/main/java/uk/gov/justice/hmpps/casenotes/notes/WriteCaseNote.kt

```diff
@@ -93,5 +93,5 @@ class WriteCaseNote(
     context.userDisplayName,
     text,
     systemGenerated ?: false,
-  )
+  ).apply { legacyId = noteRepository.getNextLegacyId() }
```

Comment:
On this path, all notes get created in the local db so no check required I guess